app/src/main/java/com/example/trashminder/utils/AlarmReceiver.kt

```diff
@@ -79,13 +89,85 @@ class AlarmReceiver : BroadcastReceiver() {
             }
         }
 
-        notifications.setRepetitiveAlarm(
-            cal.timeInMillis,
-            context,
-            cal.timeInMillis.toInt(),
-            type,
-            repetition
+        val formatter = SimpleDateFormat("dd/MM/yyyy HH:mm", Locale.FRANCE)
+        val dateString = formatter.format(Date(cal.timeInMillis))
+
+        if (idReminder != null) {
+            updateReminder(date, repetition, type, dateString, idReminder)
+
+            notifications.setRepetitiveAlarm(
+                cal.timeInMillis,
+                context,
+                cal.timeInMillis.toInt(),
+                type,
+                repetition,
+                dateString,
+                idReminder + 1
+            )
+        }
+
+        Log.d("Update", "dateString = $dateString")
+    }
+
+    private fun updateReminder(
+        date: String?,
+        repetition: String?,
+        type: String?,
+        dateString: String,
+        idReminder: Int
+    ) {
+        val trash = mutableMapOf(
+            "date" to date,
+            "id" to idReminder,
+            "repetition" to repetition,
+            "type" to type
+        )
+
+        Log.d("Update", "dateString = $dateString")
+
+        val newTrash = mutableMapOf(
+            "date" to dateString,
+            "id" to idReminder + 1,
+            "repetition" to repetition,
+            "type" to type
         )
```

Comment:
I see these 4 fields keep repeating here and there. Could this be replaced with a data class instead? And instead of passing 4 args into a function, only one would be passed?

Reply:
Also, you could make it serializable and then pass it on to intent as a string. So instead of having 
```val type = intent?.getStringExtra(REMINDER_TYPE)
        val repetition = intent?.getStringExtra(REMINDER_REPETITION)
        val date = intent?.getStringExtra(REMINDER_DATE)
        val idReminder = intent?.getIntExtra(REMINDER_ID, 0)
```
you would have `val goodVarName = intent?.getStringExtra(SERIALIZABLE_DATA_CLASS)`

Reply:
https://kotlinlang.org/docs/serialization.html#example-json-serialization